tests/dialects/test_sqlite.py

```diff
@@ -140,6 +140,19 @@ def test_sqlite(self):
             read={"snowflake": "LEAST(x, y, z)"},
             write={"snowflake": "LEAST(x, y, z)"},
         )
+        self.validate_all(
+            "SELECT STR_TO_DATE('2022-01-01', '%Y-%m-%d')",
+            write = {
+                        "mysql": "SELECT STR_TO_DATE('2022-01-01', '%Y-%m-%d')",
+                        "sqlite": "SELECT STRFTIME('%Y-%m-%d', '2022-01-01')"
+                    }
+        )
+        self.validate_all(
+            "SELECT CAST('2022-01-01' AS DATE FORMAT '%Y-%m-%d')",
+            write = {
+                "sqlite": "SELECT STRFTIME('%Y-%m-%d', '2022-01-01')"
+            }
+        )
```

Comment:
This test is incorrect, perhaps because of a misunderstanding surrounding `validate_all`. The way you've written this is like expecting the two queries to produce the same results because you're both reading and writing SQLite, but if you run them against the engine you'll notice they produce different results.